packages/wallets/src/metamask/metamaskApp.ts

```diff
@@ -0,0 +1,264 @@
+import { WalletConfig } from '../wallets.constants';
+import { WalletPage } from '../wallet.page';
+import { expect } from '@playwright/test';
+import { test, BrowserContext, Page } from '@playwright/test';
+import { MainPage, LoginPage, SettingsPage } from './pages';
+import {
+  OnboardingPage,
+  WalletOperationPage,
+  Header,
+  NetworkList,
+  OptionsMenu,
+  PopoverElements,
+  AccountMenu,
+} from './pages/elements';
+
+export class MetamaskApp implements WalletPage {
+  page: Page | undefined;
+  header: Header;
+  mainPage: MainPage;
+  settingsPage: SettingsPage;
+  loginPage: LoginPage;
+  walletOperation: WalletOperationPage;
+  onboardingPage: OnboardingPage;
+  networkList: NetworkList;
+  optionsMenu: OptionsMenu;
+  popoverElements: PopoverElements;
+  accountMenu: AccountMenu;
+
+  constructor(
+    private browserContext: BrowserContext,
+    private extensionUrl: string,
+    public config: WalletConfig,
+  ) {}
+
+  async initLocators() {
+    this.page = await this.browserContext.newPage();
+    this.header = new Header(this.page);
+    this.mainPage = new MainPage(this.page, this.extensionUrl, this.config);
+    this.settingsPage = new SettingsPage(
+      this.page,
+      this.extensionUrl,
+      this.config,
+    );
+    this.loginPage = new LoginPage(this.page, this.config);
+    this.walletOperation = new WalletOperationPage(this.page);
+    this.onboardingPage = new OnboardingPage(this.page, this.config);
+    this.networkList = new NetworkList(this.page);
+    this.optionsMenu = new OptionsMenu(this.page);
+    this.popoverElements = new PopoverElements(this.page);
+    this.accountMenu = new AccountMenu(this.page);
+  }
+
+  async navigate() {
+    await test.step('Navigate to metamask', async () => {
+      await this.initLocators();
+      await this.mainPage.openWidgetPage();
```

Comment:
i think we must to open new page before openWidgetPage. 
Because if you create object MetamaskApp once, then you will be open widget page in exist page in this.page which initialized in initLocators.

And if you will using `changeNetwork` you will close this page, and then if you would use another method from this class, you couldnt do it, because this page will be without actual context.

Reply:
I agree that process isn't ideal, but in this PR I don't want to experiment with functions logic